haystack/core/pipeline/pipeline.py

```diff
@@ -22,6 +22,19 @@ class Pipeline(PipelineBase):
     Orchestrates component execution according to the execution graph, one after the other.
     """
 
+    def _component_has_enough_inputs_to_run(self, name: str, last_inputs: Dict[str, Dict[str, Any]]) -> bool:
+        """
+        Returns True if the Component has all the inputs it needs to run.
+        """
+        instance: Component = self.graph.nodes[name]["instance"]
+        if name not in last_inputs:
+            return False
+        expected_inputs = set(instance.__haystack_input__._sockets_dict.keys())  # type: ignore
+        current_inputs = set(last_inputs[name].keys())
```

Comment:
Can't we assign a type hint to `expected_inputs` to eliminate the `type: ignore` on line 34?

Reply:
The problem is the `__haystack_input__` field. 
It would make the code more complex for no gain, the field will always be there.